processor.go

```diff
@@ -0,0 +1,235 @@
+package nakadi
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ProcessorOptions contains optional parameters that are used to create a Processor.
+type ProcessorOptions struct {
+	// The maximum number of Events in each chunk (and therefore per partition) of the stream (default: 1)
+	BatchLimit uint
+	// The number of parallel streams the Processor will use to consume events (default: 1)
+	StreamCount uint
+	// Limits the number of events that the processor will handle per minute. This value represents an
+	// upper bound, if some streams are not healthy e.g. if StreamCount exceeds the number of partitions,
+	// or the actual batch size is lower than BatchLimit the actual number of processed events can be
+	// much lower. 0 is interpreted as no limit at all (default: no limit)
+	EventsPerMinute uint
+	// The initial (minimal) retry interval used for the exponential backoff. This value is applied for
+	// stream initialization as well as for cursor commits.
+	InitialRetryInterval time.Duration
+	// MaxRetryInterval the maximum retry interval. Once the exponential backoff reaches this value
+	// the retry intervals remain constant. This value is applied for stream initialization as well as
+	// for cursor commits.
+	MaxRetryInterval time.Duration
+	// MaxElapsedTime is the maximum time spent on retries when committing a cursor. Once this value
+	// was reached the exponential backoff is halted and the cursor will not be committed.
+	CommitMaxElapsedTime time.Duration
+	// NotifyErr is called when an error occurs that leads to a retry. This notify function can be used to
+	// detect unhealthy streams. The first parameter indicates the stream No that encountered the error.
+	NotifyErr func(uint, error, time.Duration)
+	// NotifyOK is called whenever a successful operation was completed. This notify function can be used
+	// to detect that a stream is healthy again. The first parameter indicates the stream No that just
+	// regained health.
+	NotifyOK func(uint)
```

Comment:
I think it would be better to have just one function `Notify(uint,error)`  instead of two. Then, `err==nil` can indicate that this is a similar NotifyOK case.
And instead of uint as stream identifier, there can be stream identifier that . is returned from nakadi. That might help traceability. 
However, it could be done in the scope of separate issue

Reply:
I think this is a good idea but should be handled in a separate ticket: the same should be applied to `StreamOptions` where this would lead to a braking change of the API.